arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -274,11 +285,44 @@ void DatabaseFeature::collectOptions(
       ->addOption("--database.extended-names-databases",
                   "Allow most UTF-8 characters in database names. Once in use, "
                   "this option cannot be turned off again.",
-                  new options::BooleanParameter(&_extendedNamesForDatabases),
+                  new options::BooleanParameter(&_extendedNamesDatabases),
                   options::makeDefaultFlags(options::Flags::Uncommon,
                                             options::Flags::Experimental))
       .setIntroducedIn(30900);
 
+  options
+      ->addOption("--database.extended-names-collections",
+                  "Allow most UTF-8 characters in collection and index names. "
+                  "Once in use, "
+                  "this option cannot be turned off again.",
+                  new BooleanParameter(&_extendedNamesCollections),
+                  arangodb::options::makeDefaultFlags(
+                      arangodb::options::Flags::Uncommon,
+                      arangodb::options::Flags::Experimental))
+      .setIntroducedIn(31100);
+
+  options
+      ->addOption("--database.extended-names-views",
+                  "Allow most UTF-8 characters in view names. "
+                  "Once in use, "
+                  "this option cannot be turned off again.",
+                  new BooleanParameter(&_extendedNamesViews),
+                  arangodb::options::makeDefaultFlags(
+                      arangodb::options::Flags::Uncommon,
+                      arangodb::options::Flags::Experimental))
+      .setIntroducedIn(31100);
+
+  options
+      ->addOption("--database.extended-names-indexes",
+                  "Allow most UTF-8 characters in index names. "
+                  "Once in use, "
+                  "this option cannot be turned off again.",
+                  new BooleanParameter(&_extendedNamesIndexes),
+                  arangodb::options::makeDefaultFlags(
+                      arangodb::options::Flags::Uncommon,
+                      arangodb::options::Flags::Experimental))
+      .setIntroducedIn(31100);
+
```

Comment:
Why cannot we make single parameter:
1. only ascii
2. only extend database
3. extend all

Why do we need all possible combinations?
It's difficult to test, etc

Reply:
We could do this.
But with individual parameters, drivers can be adjusted piece by piece, not all or nothing.
That's why I added all the individual parameters.

Reply:
@jsteemann how would it work in case of dump/restore and/or HotBackup? One need to know all values for these prior to restore?

Reply:
No. But you simply cannot restore a dump from a deployment that has extended names set to on into a deployment that has set it to off.

Reply:
I am also happy to remove the startup options and allow extended names everywhere.
It is just that some drivers and external tools outside of our ArangoDB 3.11 release packages then could fail, if they are not adjusted. That is the only reason why we have these options in the first place, to ensure compatibility with older drivers and tools.

Reply:
I would like to have only 1 option. Just my personal favor. Old drivers should work with old ArangoDB though. Or, in case of upgrade, there is no such names initially and then it's up to user whether to turn this option on but don't create such names (to preserve old driver stability).

Reply:
We can also remove all these options and only support extended names.
Then it would be up to the users to not create objects with extended names, at least unless they have ensured all their tools support them.

Reply:
the extended database names were unstable till now, right? So one could easily change the setting to a common one enabling both?

Reply:
> We can also remove all these options and only support extended names.
Then it would be up to the users to not create objects with extended names, at least unless they have ensured all their tools support them.

This is even better.

Reply:
> the extended database names were unstable till now, right? So one could easily change the setting to a common one enabling both?

I would made it `true` by default then and bind all new logic with database-related entities (collections, views, indexes) to this option.